languages/python/exercises/concept/strings/.docs/hints.md

```diff
@@ -0,0 +1,23 @@
+# Hints
+
+### General
+
+- The [Python documentation for `str`][https://docs.python.org/3.7/library/stdtypes.html#text-sequence-type-str] is an overview of the Python `str` type.
+
+### 1. Get message from a log line
+
+ - Strings in Python have lots of convenient instance methods, like [`split()`][https://docs.python.org/3/library/stdtypes.html#str.split]
+ - There are also [methods to assist handling extra whitespace][https://docs.python.org/3.7/library/stdtypes.html#str.strip] on the ends of a string.
+
+### 2. Get log level from a log line
+
+ - Strings have methods that help [convert letters from lower to uppercase][https://docs.python.org/3/library/stdtypes.html#str.lower] and vice-versa.
+
+### 3. Reformat a log line
+
+Strings are immutable, but can be combined together to make new strings, or have elements replaced.
+Python also has a concept of string formatting, like many other languages.
+
+ - https://docs.python.org/3/library/stdtypes.html#str.join
```

Comment:
Is one of the three mentioned ways to combine strings preferred over the others? If so, it might be useful to guide the user to that method.

Reply:
There's no true preference; `str.join` is the best choice for combining a series of strings around some common delimiter ... `str.format` and f-strings are broadly similar, but there are subtle differences (the latter is more powerful, but captures from the local environment at runtime in such a manner as to make them _useless_ for template strings that need to be internationalized. Most likely we will need distinct concepts for each to cover those differences, so I'm not sure we need to state a preference here.

Reply:
Another part of the goal with listing more than one option was to make clear to the student that there are a wealth of options available, and to provide a sense of familiarity around their docs. I see the value in focusing on fewer items in order not to confuse, though.

Reply:
I understand the reasons for listing several. My main concern would be that students might get confused when they see several options, and not knowing when to use which.